### ISSUE TITLE
midtrain stuck

### ISSUE THREAD

**Issue by couldn** [karpathy/nanochat#92]
I'm stuck at midtrain, and the logs have stopped printing. I'm using 8×A100 GPUs, and it has been 12 hours already. Is this normal?

2025-10-16 19:59:50,381 - nanochat.checkpoint_manager - INFO - No model tag provided, guessing model tag: d20
2025-10-16 19:59:50,381 - nanochat.checkpoint_manager - INFO - Loading model from /root/.cache/nanochat/base_checkpoints/d20 with step 21400
2025-10-16 19:59:51,395 - nanochat.checkpoint_manager - INFO - Building model with config: {'sequence_len': 2048, 'vocab_size': 65536, 'n_layer': 20, 'n_head': 10, 'n_kv_head': 10, 'n_embd': 1280}
Tokens / micro-batch / rank: 32 x 2048 = 65,536
Tokens / micro-batch: 524,288
Total batch size 524,288 => gradient accumulation steps: 1
Scaling the LR for the AdamW parameters ∝1/√(1280/768) = 0.774597
Muon: Grouping 80 params of shape torch.Size([1280, 1280]), device cuda:0, dtype torch.float32
Muon: Grouping 20 params of shape torch.Size([1280, 5120]), device cuda:0, dtype torch.float32
Muon: Grouping 20 params of shape torch.Size([5120, 1280]), device cuda:0, dtype torch.float32

**Comment by couldn** [comment]
stuck in here

train_dataset = TaskMixture([
    SmolTalk(split="train"), # 460K rows of general conversations
    MMLU(subset="auxiliary_train", split="train"), # 100K rows of multiple choice problems drawn from ARC, MC_TEST, OBQA, RACE
    GSM8K(subset="main", split="train"), # 8K rows teaching simple math and (calculator) tool use
])

**Comment by jwnsu** [comment]
A100 80GB and -depth 20? You can try 1 GPU first, if it moves past the stuck point, increase # of GPUs incrementally (1->2->4.) It finishes pretty quick with a single GPU for me (e.g. set  --nproc_per_node=1).
